tests/tools/onert_train/test/rawdataloader.test.cc

```diff
@@ -0,0 +1,223 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <nnfw.h>
+
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <numeric>
+
+#include "../src/rawdataloader.h"
+#include "../src/nnfw_util.h"
+
+namespace
+{
+using namespace onert_train;
+
+class DataFileGenerator
+{
+public:
+  DataFileGenerator(uint32_t data_length)
+    : _data_length{data_length}, _input_file{"input.bin"}, _expected_file{"expected.bin"}
+  {
+  }
+  ~DataFileGenerator()
+  {
+    try
+    {
+      if (std::remove(_input_file.c_str()) != 0)
+      {
+        std::cerr << "Failed to remove " << _input_file << std::endl;
+      }
+      if (std::remove(_expected_file.c_str()) != 0)
+      {
+        std::cerr << "Failed to remove " << _expected_file << std::endl;
+      }
+    }
+    catch (const std::exception &e)
+    {
+      std::cerr << "Exception: " << e.what() << std::endl;
+    }
+  }
+
+  template <typename T> const std::string &generateInputData(const std::vector<T> &data)
+  {
+    generateData(_input_file, data);
+    return _input_file;
+  }
+
+  template <typename T> const std::string &generateExpectedData(const std::vector<T> &data)
+  {
+    generateData(_expected_file, data);
+    return _expected_file;
+  }
+
+private:
+  template <typename T> void generateData(const std::string &name, const std::vector<T> &data)
+  {
+    try
+    {
+      std::ofstream file(name, std::ios::binary);
+      for (uint32_t i = 0; i < _data_length; ++i)
+      {
+        for (uint32_t j = 0; j < data.size(); ++j)
+        {
+          file.write(reinterpret_cast<const char *>(&data[j]), sizeof(data[j]));
+        }
+      }
+    }
+    catch (const std::exception &e)
+    {
+      std::cerr << "Exception: " << e.what() << std::endl;
+    }
+  }
+
+private:
+  uint32_t _data_length;
+  std::string _input_file;
+  std::string _expected_file;
+};
+
+class RawDataLoaderTest : public testing::Test
+{
+protected:
+  void SetUp() override { nnfw_create_session(&_session); }
+
+  void TearDown() override { nnfw_close_session(_session); }
+
+  nnfw_session *_session = nullptr;
+};
+
+TEST_F(RawDataLoaderTest, loadDatas_1)
+{
+  const uint32_t data_length = 100;
+  const uint32_t num_input = 1;
+  const uint32_t num_expected = 1;
+  const uint32_t batch_size = 16;
+
+  // Set data tensor info
+  nnfw_tensorinfo in_info = {
+    .dtype = NNFW_TYPE_TENSOR_INT32,
+    .rank = 4,
+    .dims = {1, 2, 2, 2},
+  };
+  std::vector<nnfw_tensorinfo> in_infos{in_info};
+
+  nnfw_tensorinfo expected_info = {
+    .dtype = NNFW_TYPE_TENSOR_INT32,
+    .rank = 4,
+    .dims = {1, 1, 1, 1},
+  };
+  std::vector<nnfw_tensorinfo> expected_infos{expected_info};
+
+  // Generate test data
+  std::vector<uint32_t> in(num_input * num_elems(&in_info));
+  std::generate(in.begin(), in.end(), [] {
```

Comment:
@jyoungyun 

1) What c++ is required for this? c++17? c++20?

https://en.cppreference.com/w/cpp/algorithm/generate says `std::generate` is since c++17 or c++20.

2) Do we need `std::generate`? From the name of `RawDataLoader`, I expect the data is loaded as it is. Why do we need `std::generate`?

Reply:
I considered the huge data file. In this case, we can't load whole data file at once. So I used the std::generate function to load partial data from the file pointer.

Reply:
For 1, it seems ok. (CI passed).
For 2 (`std::generate`), I found `std::generate` is used in `rawdataloader.test.cc` for generating raw test file.

About returning `Generator`, it is confused and not clear at least for me.
Let's talk this later when the huge data case is handled.